### PR TITLE
Fix security vulnerability

### PR DESCRIPTION
**Pull request by chtyim**
This PR is still missing the jetty-server upgrade from 8 to 9, which is difficult due to a lot of incompatible changes.

### CODE REVIEW

**Comment by chtyim** [comment]
Build https://builds.cask.co/browse/CDAP-DUT6657-2

File: cdap-security/src/test/java/co/cask/cdap/security/server/ExternalAuthenticationServerTestBase.java [hunk 1]
```diff
@@ -140,8 +137,17 @@ protected void tearDown() throws Exception {
...
+  /**
+   * Returns the full URL for the given request path
+   */
+  protected URL getURL(String path) throws MalformedURLException {
+    InetSocketAddress serverAddr = server.getSocketAddress();
+    while (path.startsWith("/")) {
+      path = path.substring(1);
+    }
+    return new URL(String.format("%s://%s:%d/%s", getProtocol(),
+                                         serverAddr.getHostName(),
```
Comment (albertshau) [inline cdap-security/src/test/java/co/cask/cdap/security/server/ExternalAuthenticationServerTestBase.java]: nit: indentation

**Review by albertshau — DISMISSED** [review]
minor comments

File: cdap-security/src/test/java/co/cask/cdap/security/server/ExternalAuthenticationServerTestBase.java (hunk 1 above)
Comment (chtyim) [reply cdap-security/src/test/java/co/cask/cdap/security/server/ExternalAuthenticationServerTestBase.java]: Fixed

**Review by albertshau — APPROVED** [review]
lgtm

**Pull request merged**